floor/test/integration/stream_query_test.dart

```diff
@@ -126,5 +128,36 @@ void main() {
         expect(actual, emits(<Person>[]));
       });
     });
+
+    test('regression test streaming updates from other Dao', () async {
+      final person1 = Person(1, 'Simon');
+      final person2 = Person(2, 'Frank');
+      final dog1 = Dog(1, 'Dog', 'Doggie', person1.id);
+      final dog2 = Dog(2, 'OtherDog', 'Doggo', person2.id);
+
+      final actual = personDao.findAllDogsOfPersonAsStream(person1.id);
+      expect(
+          actual,
+          emitsInOrder(<List<Dog>>[
+            [], // initial state,
+            [dog1], // after inserting dog1
+            [dog1], // after inserting dog2
+            //[], // after removing person1. Does not work because
+            // ForeignKey-relations are not considered. (TODO)
```

Comment:
technically this can be solved without adding a query parser because we already know the foreign key relations and their effects (onUpdate/onDelete), but I'll include it in the query parser so I'll link to it.